server/pulp/server/async/app.py

```diff
@@ -93,6 +93,18 @@ def stop(self, consumer):
             self.timer_ref.cancel()
             self.timer_ref = None
 
+    def shutdown(self, consumer):
```

Comment:
So this shutdown() will be called by the consumer bootstep when it is shutting down. Can you describe specifically which situations cause this codepath to be called? I want to make sure we're only deleting the worker (and its reservations) in the right situations and none of the wrong situations.

Reply:
Only when it is being torn down entirely, i.e. this won't happen merely when the broker goes down.  That's when stop() gets called.

Reply:
OK this sounds completely right. Thanks @dralley